resources/dw/resource_hive_vw.go

```diff
@@ -0,0 +1,139 @@
+// Copyright 2024 Cloudera. All Rights Reserved.
+//
+// This file is licensed under the Apache License Version 2.0 (the "License").
+// You may not use this file except in compliance with the License.
+// You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS
+// OF ANY KIND, either express or implied. Refer to the License for the specific
+// permissions and limitations governing your use of the file.
+
+package dw
+
+import (
+	"context"
+	"github.com/cloudera/terraform-provider-cdp/cdp-sdk-go/cdp"
+	"github.com/cloudera/terraform-provider-cdp/cdp-sdk-go/gen/dw/client/operations"
+	"github.com/cloudera/terraform-provider-cdp/cdp-sdk-go/gen/dw/models"
+	"github.com/cloudera/terraform-provider-cdp/utils"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+type hiveResource struct {
+	client *cdp.Client
+}
+
+var (
+	_ resource.Resource              = (*hiveResource)(nil)
+	_ resource.ResourceWithConfigure = (*hiveResource)(nil)
+)
+
+func NewHiveResource() resource.Resource {
+	return &hiveResource{}
+}
+
+func (r *hiveResource) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	r.client = utils.GetCdpClientForResource(req, resp)
+}
+
+func (r *hiveResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_vw_hive"
+}
+
+func (r *hiveResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = hiveSchema
```

Comment:
I'd still recommend inlining the `hiveSchema` here. What is the reason for having it in a separate file @balazsgaspar @gregito ?

Reply:
I will need to add a bunch or extra params here, once I'm done if we still think it would be better to add it here, then I can

Reply:
@vcsomor having the schema in a separate place makes it easier to test what params are present and what not and also makes the resource shorter therefore more readable